### PR TITLE
New cop: RSpec/EmptyLineAfterExampleGroup

### PR DESCRIPTION
**Pull request by bquorning**
This cop will verify that there is an empty line after hook blocks (before, after, around, etc.)

Ref: #272

This cop is very much inspired by `EmptyLineAfterSubject` and `EmptyLineAfterHook` (#632) and perhaps we can extract some common parts to e.g. a module.

---

Before submitting the PR make sure the following are checked:

* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [changelog](https://github.com/rubocop-hq/rubocop-rspec/blob/master/CHANGELOG.md) if the new code introduces user-observable changes.
* [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

If you have created a new cop:

* [x] Added the new cop to `config/default.yml`.
* [x] The cop documents examples of good and bad code.
* [x] The tests assert both that bad code is reported and that good code is not reported.

### CODE REVIEW

File: lib/rubocop/cop/rspec/empty_line_after_example_group.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        def on_block(node)
+          return unless example_group(node)
+          return if node.parent && node.equal?(node.parent.children.last)
+
+          return if next_line(node).blank?
+
+          add_offense(
+            node,
+            location: node.loc.end,
+            message: format(MSG, example_group: node.send_node.method_name)
```
Comment (Darhazer) [inline lib/rubocop/cop/rspec/empty_line_after_example_group.rb]: method_name works for block_nodes as well, so no need for `send_node`

File: lib/rubocop/cop/rspec/empty_line_after_example_group.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks if there is an empty line after example group blocks.
+      #
+      # @example
+      #   # bad
+      #   RSpec.describe Foo do
+      #   end
+      #   RSpec.describe Bar do
```
Comment (Darhazer) [inline lib/rubocop/cop/rspec/empty_line_after_example_group.rb]: those seems as offenses for the multipledescribes cop, better add examples for few different describes in a single top-level describe ;)

**Comment by Darhazer** [comment]
> This cop is very much inspired by EmptyLineAfterSubject and EmptyLineAfterHook (#632) and perhaps we can extract some common parts to e.g. a module.

Sure we can. I would extract the check if the next line is blank, as well whether the node is the one one in a block and perhaps the autocorrect
```
# common checks
return if node.parent && node.equal?(node.parent.children.last)

return if next_line(node).blank?
```

File: lib/rubocop/cop/rspec/empty_line_after_example_group.rb (hunk 2 above)
Comment (bquorning) [reply lib/rubocop/cop/rspec/empty_line_after_example_group.rb]: That is a very good point, actually. I’ll change here, and all the specs.

**Comment by bquorning** [comment]
Fixed the issues mentioned in a separate commit for easier review. I’ll squash before merging :-)

Then, I’ll take a stab a extracting shared logic from the “empty line after …” cops.

**Pull request merged**